specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2023-07-01-preview/virtualMachineInstances.json

```diff
@@ -0,0 +1,1545 @@
+              }
+            },
+            "processors": {
+              "type": "integer",
+              "format": "int32",
+              "description": "number of processors for the virtual machine instance"
+            },
+            "memoryMB": {
+              "type": "integer",
+              "format": "int64",
+              "description": "RAM in MB for the virtual machine instance"
+            },
+            "dynamicMemoryConfig": {
+              "type": "object",
+              "properties": {
+                "maximumMemoryMB": {
+                  "type": "integer",
+                  "format": "int64"
+                },
+                "minimumMemoryMB": {
+                  "type": "integer",
+                  "format": "int64"
+                },
+                "targetMemoryBuffer": {
+                  "type": "integer",
+                  "description": "Defines the amount of extra memory that should be reserved for a virtual machine instance at runtime, as a percentage of the total memory that the virtual machine instance is thought to need. This only applies to virtual systems with dynamic memory enabled. This property can be in the range of 5 to 2000.",
+                  "format": "int32"
+                }
+              }
+            }
+          },
+          "description": "HardwareProfile - Specifies the hardware settings for the virtual machine instance."
+        },
+        "networkProfile": {
+          "type": "object",
+          "properties": {
+            "networkInterfaces": {
+              "type": "array",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "id": {
+                    "type": "string",
+                    "description": "ID - Resource Id of the network interface"
+                  }
+                }
+              },
+              "description": "NetworkInterfaces - list of network interfaces to be attached to the virtual machine instance"
+            }
+          },
+          "description": "NetworkProfile - describes the network configuration the virtual machine instance"
+        },
+        "osProfile": {
+          "type": "object",
+          "properties": {
+            "adminPassword": {
+              "type": "string",
+              "x-ms-secret": true,
+              "description": "AdminPassword - admin password",
+              "x-ms-mutability": [
+                "create"
+              ]
+            },
+            "adminUsername": {
+              "type": "string",
+              "description": "AdminUsername - admin username"
+            },
+            "computerName": {
+              "type": "string",
+              "description": "ComputerName - name of the compute"
+            },
+            "linuxConfiguration": {
+              "type": "object",
+              "properties": {
+                "disablePasswordAuthentication": {
+                  "type": "boolean",
```

Comment:
>boolean [](http://example.com/codeflow?start=27&length=7)

Consider enum over boolean for best practice

Reply:
Please see [this comment](https://github.com/Azure/azure-rest-api-specs-pr/pull/13356#issuecomment-1640658444) from our RPSaaSMaster PR explaining why we would not like to do that.

Reply:
Resolving this since @zizw123 said he didn't have additional feedback.